### PR TITLE
[Netlify] New commands and improvements

### PR DESCRIPTION
**Pull request by jasonbarry**
## Description

This PR creates a few new commands for the Netlify extension, and improves upon existing commands. 

- Added `Find Local Sites` command.
- Added `Invite Team Members` command.
- Added `Search Audit Log` command.
- Added `Search Docs` command.
- Renamed `View Sites` -> `Search Sites`, and added site and deploy metadata in a detail view.
- Renamed `View Domains` -> `Search Domains`, and added the ability to register a new domain.
- Renamed `View Members` -> `Search Team Members`, and added Git contributors and reviewers to the list.
- Expanded README to include screenshots and helpful instructions.

## Screencast

https://www.loom.com/share/7c51ca1f4ec540eab7dd2f1836e527a6

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/netlify/src/components/deploys.tsx
Comment (thomaslombart) [inline extensions/netlify/src/components/deploys.tsx:42]: Note that you can use some hooks from `@raycast/utils` to make async stuff easier to manage: either [useFetch](https://developers.raycast.com/utilities/react-hooks/usefetch) or [useCachedPromise](https://developers.raycast.com/utilities/react-hooks/usecachedpromise). In your case, `useFetch` should be enough.

File: extensions/netlify/src/components/deploys.tsx
Comment (thomaslombart) [inline extensions/netlify/src/components/deploys.tsx:134]: Was that comment intended?

File: extensions/netlify/src/invite-team-members.tsx
Comment (thomaslombart) [inline extensions/netlify/src/invite-team-members.tsx:43]: Note that you can also use the [useForm](https://developers.raycast.com/utilities/react-hooks/useform) hook from the `@raycast/utils` package. This makes forms easier to manage as well, especially for validators.

**Review by thomaslombart — COMMENTED** [review]
Thanks for putting up this PR! 🚀

I found a couple of bugs while testing the extension:

I don't have any domains on my Netlify test account, and I get this empty state when searching for domain names:

netlify 2023-01-19 at 15 28 48

When I also search for a new domain name, I get the following error:

netlify 2023-01-19 at 15 50 33

I have two consecutive separators on one of the test sites deployed:

netlify 2023-01-19 at 15 49 30

I guess something is conditionally shown for the use case above. 

Finally, I'd suggest providing a shortcut to open the GitHub Profile when searching for a team member:

netlify 2023-01-19 at 15 51 51

Finally, I've said it in some comments, but generally, I'd suggest using `@raycast/utils` hooks throughout the codebase to abstract most of the async/forms management stuff.

File: extensions/netlify/src/components/deploys.tsx
Comment (jasonbarry) [reply extensions/netlify/src/components/deploys.tsx:42]: How can `useFetch` be used if authorization headers need to be sent with the request?

Comment (thomaslombart) [reply extensions/netlify/src/components/deploys.tsx:42]: `useFetch` options extends [RequestInit](https://github.com/nodejs/undici/blob/v5.7.0/types/fetch.d.ts#L103-L117) so you should be able to send authorization headers like the example below:

```ts
const { isLoading, data, error } = useFetch<Advice>(url, {
  headers: {
    // ...
  },
  // ...
});
```

Comment (jasonbarry) [reply extensions/netlify/src/components/deploys.tsx]: Thanks for the suggestion @thomaslombart! I ended up going with `usePromise`, and was able to delete a lot of redundant code. All of the `usePromise` changes are contained in https://github.com/raycast/extensions/pull/4270/commits/7bcde176a05b40b6ddb8bcf63a2d607ca8725610.

**Comment by thomaslombart** [comment]
Hey @jasonbarry, is this ready to review again? 😊

**Comment by jasonbarry** [comment]
Yes it is @thomaslombart!

**Review by thomaslombart — COMMENTED** [review]
Looks good to me! Any reasons why you used `usePromise` instead of `useCachedPromise`? 😊

**Comment by jasonbarry** [comment]
> Looks good to me! Any reasons why you used usePromise instead of useCachedPromise? 😊 

The API seems fast enough that I didn't see the need to deal with stale content / revalidation. I'll keep it as a consideration for slow / low-update-frequency endpoints in the future!

**Comment by jasonbarry** [comment]
@thomaslombart Anything else needed before approval/merge? 😀

**Comment by thomaslombart** [comment]
Hey @jasonbarry, yes @peduarte would like to take a look before merging. Sorry, I forgot to let you know ☺️

**Review by peduarte — APPROVED** [review]
@jasonbarry Yoooo dude, this is 🔥🔥🔥

Massive improvements. Really great work. 

A couple of notes, non-blocking
- Could we make it so the "Search Docs" Command doesnt require an API key? 
- I think a Menu Bar Command would be really nice as well, where it could show the list of sites, and status of deployments — something to keep in mind for a next version!

@thomaslombart thanks for your review!

**Comment by jasonbarry** [comment]
@peduarte thank you, it was fun to write! 

 > Could we make it so the "Search Docs" Command doesnt require an API key?

I think so, would this entail moving the API key preference under each command in package.json except docs search? We could do this for the Find Local Sites command too. I can keep that in mind for a subsequent PR. 

> I think a Menu Bar Command would be really nice as well, where it could show the list of sites, and status of deployments — something to keep in mind for a next version! 

Already planned 😊 coming soon!

**Comment by peduarte** [comment]
@jasonbarry Ok, great. Excellent work!

**Pull request merged**
